pkg/controller/controller.go

```diff
@@ -838,6 +857,14 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 		},
 	}
 
+	// Set AnnDeletionSecretRefName and namespace
+
+	if result.provDeletionSecrets != nil {
```

Comment:
@humblec 

Should we consider adding empty string for "volume.kubernetes.io/provisioner-deletion-secret-name" and "volume.kubernetes.io/provisioner-deletion-secret-namespace" annotations if Secret is not used (and making `Delete` not fail and skip adding credentials in this case)?

The current logic falls back to get name and namespace for Secret from StorageClass if these annotations are not found. However, it can't distinguish whether lack of annotations is caused by (1) no Secret required or (2) provisioned by old provisioner. As a result, volumes without Secret will fail to be deleted when StorageClass is not found, even after new provisioner is used on deploy.

Reply:
Nice catch @mkimuram ! , updated to accomodate that condition too..